packages/EasyTest/src/Traits/MessengerAssertionsTrait.php

```diff
@@ -206,8 +220,58 @@ private static function runMessengerWorker(string $transportName, string $busNam
             $messageLimitListener = new StopWorkerOnMessageLimitListener($messagesCount);
             $eventDispatcher->addSubscriber($messageLimitListener);
 
-            $worker = new Worker(['async' => $asyncTransport], $messageBus, $eventDispatcher);
+            $worker = new Worker(['async' => $transport], $messageBus, $eventDispatcher);
             $worker->run();
         }
+
+        if (\count((array)$transport->get()) > 0) {
+            throw new \RuntimeException('Unable to consume all messages from async transport.');
+        }
+
+        foreach ($transport->getRejected() as $envelope) {
+            /** @var \Symfony\Component\Messenger\Stamp\ErrorDetailsStamp $errorDetailsStamp */
+            foreach ($envelope->all(ErrorDetailsStamp::class) as $errorDetailsStamp) {
+                foreach ($expectedExceptions as $key => $expectedExceptionPair) {
```

Comment:
Do we really need the expected exceptions feature? Is not exposing of occurred errors enough?

Reply:
We have tests where we expect that message will be failed.

Reply:
Explanation added in cod comments.